.travis.yml

```diff
@@ -1,46 +1,38 @@
 language: php
 
-addons:
-  apt:
-    packages:
-      - libxml2-utils
-
-php:
-  - 7.1
-  - 7.2
-  - 7.3
-
-matrix:
-  fast_finish: true
-
-env:
-  matrix:
-    - DEPENDENCIES="high"
-    - DEPENDENCIES="low"
-  global:
-    - DEFAULT_COMPOSER_FLAGS="--no-interaction --no-ansi --no-progress --no-suggest"
-
-before_install:
-  - ./tools/composer clear-cache
-
-install:
-  - if [[ "$DEPENDENCIES" = 'high' ]]; then travis_retry ./tools/composer update $DEFAULT_COMPOSER_FLAGS; fi
-  - if [[ "$DEPENDENCIES" = 'low' ]]; then travis_retry ./tools/composer update $DEFAULT_COMPOSER_FLAGS --prefer-lowest; fi
-
-before_script:
-  - echo 'zend.assertions=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
-  - echo 'assert.exception=On' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
-
-script:
-  - ./phpunit --coverage-clover=coverage.xml
-  - ./phpunit --configuration ./build/travis-ci-fail.xml > /dev/null; if [ $? -eq 0 ]; then echo "SHOULD FAIL"; false; else echo "fail checked"; fi;
-  - xmllint --noout --schema phpunit.xsd phpunit.xml
-  - xmllint --noout --schema phpunit.xsd tests/_files/configuration.xml
-  - xmllint --noout --schema phpunit.xsd tests/_files/configuration_empty.xml
-  - xmllint --noout --schema phpunit.xsd tests/_files/configuration_xinclude.xml -xinclude
-
-after_success:
-  - bash <(curl -s https://codecov.io/bash)
+jobs:
+  include:
+    - &COLLECT_CODE_COVERAGE
+
+      stage: Collect Code Coverage
+
+      php: 7.1
+
+      if: type = pull_request
```

Comment:
This will only run for pull requests, so we can remove the overhead to the minimum required (when someone without _write_ permissions opens a pull request, we want to see how this affects code coverage).